src/pages/iou/steps/MoneyRequestConfirmPage.js

```diff
@@ -77,7 +78,27 @@ function MoneyRequestConfirmPage(props) {
     );
     const isManualRequestDM = props.selectedTab === CONST.TAB.MANUAL && iouType.current === CONST.IOU.MONEY_REQUEST_TYPE.REQUEST;
 
+    const navigateBack = () => {
+        let fallback;
+        if (reportID.current) {
+            fallback = ROUTES.MONEY_REQUEST.getRoute(iouType.current, reportID.current);
+        } else {
+            fallback = ROUTES.MONEY_REQUEST_PARTICIPANTS.getRoute(iouType.current);
+        }
+        Navigation.goBack(fallback);
+    };
+
     useEffect(() => {
+        if (props.iou.receiptPath && props.iou.receiptSource) {
+            FileUtils.readFileAsync(props.iou.receiptPath, props.iou.receiptSource).then((receipt) => {
+                if (!receipt) {
+                    navigateBack();
+                } else {
+                    setReceiptFile(receipt);
```

Comment:
Why don't we set the `receipt.state` here?

Reply:
@mollfpr I updated